Sources/HummingbirdRouter/RouteController.swift

```diff
@@ -0,0 +1,106 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2023-2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Hummingbird
+
+// MARK: - RouteController
+
+/// A type that defines a body of routes.
+///
+/// You can create custom controllers by declaring types that conform to the `RouteController`
+/// protocol. Implement the required ``RouteController/body`` computed property and provide a
+/// type for the controller's `Context`.
+///
+///     struct UsersController {
+///         typealias Context = BasicRouterRequestContext
+///
+///         var body: some RouterMiddleware<Context> {
+///             RouteGroup("/users") {
+///                 ...
+///             }
+///         }
+///     }
+///
+/// Assemble the controller's body by combining one or more components together to create
+/// a route. By nesting controllers within each other, you can compose large and complex
+/// routes from smaller more mangeable components.
+public protocol RouteController where Context == Body.Context {
+    associatedtype Context: RouterRequestContext
+    associatedtype Body: RouterMiddleware
+    @RouteBuilder<Context> var body: Body { get }
+}
+
+// MARK: - MiddlewareFixedTypeBuilder + RouteController
+
+/// Middleware stack result builder
+///
+/// Generates a middleware stack from the elements inside the result builder. The input,
+/// context and output types passed through the middleware stack are fixed and cannot be changed.
+extension MiddlewareFixedTypeBuilder {
+    public static func buildExpression<C0: RouteController>(_ c0: C0) -> C0.Body where C0.Body.Input == Input, C0.Body.Output == Output, C0.Body.Context == Context {
```

Comment:
I've written all the other MiddlewareFixedTypeBuilder functions using MiddlewareProtocol. It would be great if you could continue that.

Reply:
I'm not sure I follow the ask here. Given that we decided not to have the `RouteController` conform to `MiddlewareProtocol`, I believe I have to conform the build expression's generics to `RouteController` in order to compose them together.

I might be misunderstanding the ask though.

Reply:
I'm not sure I follow the ask here. Given that `RouteController` itself doesn't conform to `MiddlewareProtocol` I think I need to constrain the generic to `RouteController`. I might be misunderstanding the ask though.

Reply:
I updated the `RouteController` to have a body that is a `MiddlewareProtocol` and I also changed it to use `@MiddlewareFixedTypeBuilder` instead. Maybe that's what you were referring to?